### PR TITLE
Fix errors reported by static analyzers.

### PR DESCRIPTION
**Pull request by ADmad**

### CODE REVIEW

File: src/Http/ServerRequest.php [hunk 1]
```diff
@@ -1268,11 +1268,10 @@ public function getData(?string $name = null, mixed $default = null): mixed
         if ($name === null) {
             return $this->data;
         }
-        if (!is_array($this->data) && $name) {
+        if (!is_array($this->data)) {
```
Comment (othercorey) [inline src/Http/ServerRequest.php:1271]: We no longer care about `''` here, right? It shouldn't be used for anything meaningful.

Comment (ADmad) [reply src/Http/ServerRequest.php:1271]: Don't see a practical case of having the data array with empty string as key. At least not possible through a form submission.

Comment (ADmad) [reply src/Http/ServerRequest.php:1271]: Err if data is non-array then the name becomes irrelevant anyway since `Hash::extract()` can't be used anymore.

Comment (othercorey) [reply src/Http/ServerRequest.php:1271]: agreed.

**Review by markstory — COMMENTED** [review]
Looks good to me.

**Pull request merged**